src/core/providers/ogr/qgsogrproviderconnection.cpp

```diff
@@ -315,16 +315,19 @@ void QgsOgrProviderConnection::createVectorTable( const QString &schema,
   opts[ QStringLiteral( "driverName" ) ] = QString( GDALGetDriverShortName( hDriver ) );
   QMap<int, int> map;
   QString errCause;
+  QString createdLayerName;
   Qgis::VectorExportResult errCode = QgsOgrProvider::createEmptyLayer(
                                        uri(),
                                        fields,
                                        wkbType,
                                        srs,
                                        overwrite,
                                        &map,
+                                       createdLayerName,
                                        &errCause,
                                        &opts
                                      );
+  // TODO we need some way to hand createdLayerName back to caller, as it may differ from the requested name...
```

Comment:
Is that something you're planning to address?

Reply:
No, it's too tricky to do without breaking stable API